codes/3_coding.py

```
@@ -16,14 +17,15 @@
 args    = parser.parse_args()
-client = OpenAI(api_key = os.environ["OPENAI_API_KEY"])
 
 paper_name = args.paper_name
 gpt_version = args.gpt_version
 pdf_json_path = args.pdf_json_path
 output_dir = args.output_dir
 output_repo_dir = args.output_repo_dir
 
+agent = Agent(gpt_version)
+
 with open(pdf_json_path) as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)